### PR TITLE
job.md: add section on suspended jobs

### PR DESCRIPTION
**Pull request by adtac**
KEP issue: https://github.com/kubernetes/enhancements/issues/2232
KEP: https://github.com/kubernetes/enhancements/tree/master/keps/sig-apps/2232-suspend-jobs
Implementation: https://github.com/kubernetes/kubernetes/pull/98727

/sig apps
/hold till implementation is merged
/cc @ahg-g @alculquicondor @soltysh

### CODE REVIEW

File: content/en/docs/concepts/workloads/controllers/job.md [hunk 1]
```diff
@@ -374,6 +375,107 @@ Here, `W` is the number of work items.
...
+  Normal  SuccessfulCreate  11m   job-controller  Created pod: myjob-hlrpl
+  Normal  SuccessfulCreate  11m   job-controller  Created pod: myjob-gmszl
+  Normal  SuccessfulDelete  11m   job-controller  Deleted pod: myjob-gmszl
+  Normal  SuccessfulDelete  11m   job-controller  Deleted pod: myjob-hlrpl
+  Normal  SuccessfulDelete  11m   job-controller  Deleted pod: myjob-4v5gk
+  Normal  Suspended         11m   job-controller  Job suspended
+  Normal  SuccessfulCreate  3s    job-controller  Created pod: myjob-jvb44
+  Normal  SuccessfulCreate  3s    job-controller  Created pod: myjob-vg7x9
+  Normal  SuccessfulCreate  3s    job-controller  Created pod: myjob-tjbgv
+  Normal  Resumed           3s    job-controller  Job resumed
```
Comment (adtac) [inline content/en/docs/concepts/workloads/controllers/job.md:467]: The way the Job controller works means that the pod creation/deletion events are emitted before status condition-related events. Not sure if the order is an issue even though all of these events are emitted at the same time.

**Comment by celestehorgan** [comment]
Hi @adtac,

Is this change a part of the 1.21 release? If so, it needs to be PRed against the 1.21 branch of this repository.

**Comment by adtac** [comment]
@celestehorgan I see. I thought we merge into master until the 1.21 docs freeze? I didn't see any PRs opened against the dev-1.21 branch, so I figured targeting master should be fine :sweat_smile: 

Is this because https://github.com/kubernetes/website/pull/26404 was recently merged?

**Comment by alculquicondor** [comment]
master is what is live. Once 1.21 get's released, the 1.21 branch gets merged to master

**Comment by reylejano** [comment]
@adtac Release 1.21 Docs lead here. As this k/website PR is associated with kubernetes/enhancements#2232 which is targeted for 1.21, please change the target branch to `dev-1.21`

**Comment by adtac** [comment]
Closing in favour of https://github.com/kubernetes/website/pull/26466 (I don't know how to change the target branch of an already-open PR)

**Pull request closed without merging**